posthog/batch_exports/http.py

```diff
@@ -183,11 +187,64 @@ class Meta:
         model = BatchExportDestination
         fields = ["type", "config"]
 
-    def create(self, validated_data: dict) -> BatchExportDestination:
+    def create(self, validated_data: collections.abc.Mapping[str, typing.Any]) -> BatchExportDestination:
         """Create a BatchExportDestination."""
         export_destination = BatchExportDestination.objects.create(**validated_data)
         return export_destination
 
+    def validate(self, data: collections.abc.Mapping[str, typing.Any]) -> collections.abc.Mapping[str, typing.Any]:
+        """Validate the destination configuration based on workflow inputs.
+
+        Ensure that the submitted destination configuration passes the following checks:
+        * Does NOT contain fields that do not exist in workflow inputs.
+        * Contains all required fields as defined by workflow inputs.
+        * Provided values match types required by workflow inputs.
+
+        Raises:
+            A `serializers.ValidationError` if any of these checks fail.
+        """
+        export_type, config = data["type"], data["config"]
+        _, workflow_inputs = DESTINATION_WORKFLOWS[export_type]
+        base_field_names = {field.name for field in dataclasses.fields(BaseBatchExportInputs)}
+        workflow_fields = dataclasses.fields(workflow_inputs)
+        destination_fields = {field for field in workflow_fields if field.name not in base_field_names}
+
+        extra_fields = config.keys() - {field.name for field in destination_fields} - base_field_names
+        if extra_fields:
+            str_fields = ", ".join(f"'{extra_field}'" for extra_field in sorted(extra_fields))
+            raise serializers.ValidationError(f"Configuration has unknown field/s: {str_fields}")
+
+        for destination_field in destination_fields:
+            is_required = (
+                destination_field.default == dataclasses.MISSING
+                and destination_field.default_factory == dataclasses.MISSING
+            )
+            if destination_field.name not in config:
+                is_patch = self.context["request"].method == "PATCH"
+                if is_required and not is_patch:
+                    # When patching we expect a partial configuration. So, we don't
+                    # error on missing required fields.
+                    raise serializers.ValidationError(
+                        f"Configuration missing required field: '{destination_field.name}'"
+                    )
+                else:
+                    continue
+
+            config_value = config[destination_field.name]
+            field_type = destination_field.type
+
+            if not isinstance(field_type, type):
+                # `dataclasses.Field.type` could be something we can't work with.
+                # TODO: Validate these ones too?
```

Comment:
suppose it would be good to validate these ones too in future

Reply:
It's tricky as this could be `str` if doing something like:
```
@dataclasses.dataclass
class MyData:
    f: "MyCustomType" = ...
```

Which means we would need to resolve `"MyCustomType"` somehow.

Regardless, I don't think we are doing this at the moment, just something to keep in mind.

Primitive types should all be validated always.